libs/core/src/main/java/org/opensearch/core/tasks/TaskId.java

```diff
@@ -106,6 +115,23 @@ public static TaskId readFromStream(StreamInput in) throws IOException {
         return new TaskId(nodeId, in.readLong());
     }
 
+    /**
+     * Read a {@linkplain TaskId} from a byte array. {@linkplain TaskId} has this rather than the usual constructor that takes a
+    * {@linkplain byte[]} so we can return the {@link #EMPTY_TASK_ID} without allocating.
+    */
+    public static TaskId readFromBytes(byte[] in) throws IOException {
+        TaskIdProto.TaskId taskIdProto = TaskIdProto.TaskId.parseFrom(in);
```

Comment:
This approach forces us to always parse a bytestream for the "most common" empty task ID.  

Can we have a single byte boolean (basically the "OptionalWriteable" implementation) here which directly shortcuts to `EMPTY_TASK_ID` instead of always deserializing some number of bytes > 1?